starknet-rpc-test/get_transaction_by_blockid_and_index.rs

```diff
@@ -0,0 +1,139 @@
+    let rpc = madara.get_starknet_client();
+
+    assert_matches!(
+        rpc
+        .get_transaction_by_block_id_and_index(
+            BlockId::Number(1),
+            0
+        )
+        .await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code:
+MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::BlockNotFound     );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn fail_out_of_block_index(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    assert_matches!(
+            rpc
+            .get_transaction_by_block_id_and_index(
+                BlockId::Tag(BlockTag::Latest),
+                0
+            )
+            .await,
+            Err(StarknetProviderError(StarknetErrorWithMessage { code:
+    MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::InvalidTransactionIndex
+        );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn work_ok_by_compare_with_get_block_with_tx(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    let signer = LocalWallet::from(SigningKey::from_secret_scalar(FieldElement::from_hex_be(SIGNER_PRIVATE).unwrap()));
+    let argent_account_address = FieldElement::from_hex_be(ARGENT_CONTRACT_ADDRESS).expect("Invalid Contract Address");
+    let account = SingleOwnerAccount::new(rpc, signer, argent_account_address, chain_id::TESTNET);
+
+    madara.create_empty_block().await?;
+
+    let execution_1 = account.transfer_tokens(
+        argent_account_address,
+        FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
+        None,
+    );
+
+    let execution_2 = account
+        .transfer_tokens(
+            FieldElement::from_hex_be(TEST_CONTRACT_CLASS_HASH).expect("Invalid Contract Address"),
+            FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
+            None,
+        )
+        .nonce(FieldElement::ONE)
+        .max_fee(FieldElement::from_hex_be("0xDEADB").unwrap());
+
+    madara
+        .create_block_with_txs(vec![TransactionEnum::Execution(execution_1), TransactionEnum::Execution(execution_2)])
+        .await?;
+
+    let tx_1 = rpc.get_transaction_by_block_id_and_index(BlockId::Tag(BlockTag::Latest), 0).await?;
+    let tx_2 = rpc.get_transaction_by_block_id_and_index(BlockId::Tag(BlockTag::Latest), 1).await?;
+
+    let block_with_txs = rpc.get_block_with_txs(BlockId::Tag(BlockTag::Latest)).await?;
+
+    assert_matches!(tx_1, Transaction::Invoke(InvokeTransaction::V1(InvokeTransactionV1 {
+        nonce,
+        sender_address,
+        transaction_hash,
+        ..
+     })) if nonce == FieldElement::ZERO 
+            && sender_address == argent_account_address
+            && Some(&transaction_hash) == get_transaction_hash_from_block_with_txs(&block_with_txs, 0));
```

Comment:
Do it in another assert

```rust
assert_matches(
get_transaction_hash_from_block_with_txs(block_with_txs.transactions[index],
tx_1,
)
```

The way you did start to make it difficult to read